File: 4168-mirror-distance-of-an-integer/mirror-distance-of-an-integer.js

var mirrorDistance = function (n) {
    const getReverse = () => {
        let rev = 0;
        while (n > 0) {
            rev = rev * 10 + n % 10
            n = Math.floor(n / 10)
        }
        return rev
    }
    return Math.abs(n - getReverse(n))
};

File: 4168-mirror-distance-of-an-integer/mirror-distance-of-an-integer.cpp
class Solution {
public:
    int mirrorDistance(int n) {
        // Tc=(long(n)) for reversing 
        // Sc = O(log(n)) storing string version of n
        string s = to_string(n);
        reverse(begin(s), end(s));

        int rev = stoi(s);
        return abs(n - rev);
    }
};